spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/simple/DiscoveryClientAutoConfigurationDefaultTests.java

```diff
@@ -3,36 +3,40 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.cloud.client.discovery.composite.CompositeDiscoveryClient;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.test.context.junit4.SpringRunner;
 
+import java.util.List;
+
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * DiscoveryClient implementation defaults to {@link SimpleDiscoveryClient}
  * @author Biju Kunjummen
  */
 @RunWith(SpringRunner.class)
-@SpringBootTest(classes = DiscoveryClientAutoConfigurationDefaultTests.App.class)
+@SpringBootTest(classes = DiscoveryClientAutoConfigurationDefaultTests.Config.class)
 public class DiscoveryClientAutoConfigurationDefaultTests {
 
 	@Autowired
-	DiscoveryClient discoveryClient;
+	private List<DiscoveryClient> discoveryClients;
```

Comment:
Yes, it does @spencergibb, I have marked the `CompositeDiscoveryClient` as primary and auto-configuring the `SimpleDiscoveryClient`, a different test captures that behavior. I will restore this test to show that the defaults are maintained